### ISSUE TITLE
Idea: Annotations / Labeling of content

### ISSUE THREAD

**Issue by ThisIsMissEm** [swicg/activitypub-trust-and-safety#4]
- self-labeling of content
- moderator-labeling of content
- annotations vs hashtag-style labels
- having followable actors that apply labels to content (this is very likely annotations, not labels)
- standardised vocabularies of labels, e.g., [DTSP](https://connect.iftas.org/library/iftas-documentation/shared-vocabulary-labels/)

Potentially building off the [Web Annotations Data Model](https://www.w3.org/TR/annotation-model/)

Needs to figure out the different use cases and the different mechanisms that would be necessary.

**Comment by ThisIsMissEm** [comment]
Original discussion about a community notes alternative is here: https://github.com/w3c/activitypub/issues/464

**Comment by gobengo** [comment]
> Potentially building off the [Web Annotations Data Model](https://www.w3.org/TR/annotation-model/)

Yes! Instead of just trying to categorize/moderate whole social posts, it would be great to empower mods to annotate the exact parts of contents that fit various classifications. This would make for better training data for auto classification. More importantly, UIs and processes built around this could more easily help educate folks on acceptable policy-based norms within a community, which in some online community research shows is super important iirc from some old research from https://coralproject.net/research/

**Comment by jfinkhaeuser** [comment]
I don't necessarily disagree that annotations of specific parts are useful, it seems like this can be added later as a refinement of annotating a post.

If annotations are considered a list of objects, which each contain an author and e.g. an item from a vocabulary such as discussed in #15, then adding additional keys to refer to text parts or media parts as in #16 would be an easy upgrade.

I mention this mostly because I'm pretty sure a lot could already be achieved with the simpler solution. It would be sad to see a lack of progress due to waiting for something more complete.

At least, I do not currently see how it would be necessary to wait.

**Comment by bobwyman** [comment]
I strongly support the incorporation of annotations into the AS/AP data model. Ideally, any AP actor would be able to create an object which is an `AnnotationOf` any other object. In many instances, publishing such an object would have similar side-effects as publishing something `InReplyTo` something. (i.e. there would be annotation collections on objects, actors would have collections of annotations, etc.) 

It should be understood that an Annotation is NOT a reply, rather it is a statement about some object rather than a response to its creator.

**Comment by ShadowJonathan** [comment]
Apologies for the non-technical approach, but I think that this - having user-attached labels/annotations/freeform string fields - would be very good for the culture mix of the fediverse, as this will allow many many (currently public) disputes, announcements, and other approaches of "pointing" to users (either positive or negative) to happen with more direct and subtle input from the originator. And, the fact that the receiving user needs to explicitly agree/follow the originating labeller/annotator will make the relationship more direct, instead of publications circulating, and losing context due to context collapse.

There has been a pervasive problem of burnout, both due to boy-crying-wolf and overload of these kinds of posts, people losing fedi entirely over it, or starting to get traumatic responses, as this is often then the topic of the week, and extremely emotionally exhausting for everyone involved.

Annotations would allow the tension of these "earth quake" moments to be more gradually addressed, socially, and while I can argue that putting scuttlebutt in more private/deeper corners can have negative effects, I believe that the (unintentional) experiment of public "fedi meta" discourse over the last couple of years have proven that the public route consolidates exhaustion and energy demand as much as it does its information.

**Comment by ThisIsMissEm** [comment]
Statement from @gobengo on the [public-swicg mailing list](https://lists.w3.org/Archives/Public/public-swicg/2025Jan/0070.html):

> tl;dr: You can already do web annotation via ActivityPub by sending objects like {"type":"Annotation","@context":"http://www.w3.org/ns/anno.jsonld"} to your ActivityPub outbox via HTTP POST.
> 
> sadly there are too many 'ActivityPub Implementations' that are not "[ActivityPub Conformant Servers](https://www.w3.org/TR/activitypub/#specification-profiles)" and do not even try to implement the 'object creation' requirements of ActivityPub.
> But if they haven't implemented the already-standard way of creating annotations, we shouldn't make more new alternatives, we should encourage folks to do what has been standardized for 5+ years.
> 
> imho there is no need for an `Annotate` activity.
> 
> You can already POST an `Annotation` to an ActivityPub outbox. The ActivityPub outbox MUST accept it and then MUST wrap it in a `Create` activity.
> 
> It's not an accident that, other than a small media type parameter, this is exactly how W3C's Web Annotation Protocol says to [create an annotation](https://www.w3.org/TR/annotation-protocol/#create-a-new-annotation) in a collection.
> 
> We don't need an activity for every kind of thing, e.g. 'Annotate'. There are other vocabularies for things, e.g. the [Web Annotation Data Model](https://www.w3.org/TR/annotation-model/) and w3c protocols for it. ActivityPub already provides activities to describe the lifecycle of things, and requirements that an outbox server wrap unexpected things in a `Create` activity to start logging those lifecycle events.
> 
> New vocabulary and behaviors are not at all necessary, and will only fragment efforts even more.
> 
> The Web Annotation Data Model is already sufficient to be used as the request body of the already-standard mechanism in ActivityPub to create things: just POST it to your ActivityPub outbox (which *MUST* wrap it in 'Create' if its not an Activity).
> 
> It would be very confusing if, in addition to the already standard {type:'Create',object:{type:'Annotation'}}, there were a bunch of {type:'Annotate',object:{type:'Note'}}.
> 
> This is a good time to remember that there is nothing in ActivityPub that requires that oubox submission `type` be defined by AS2
> and that `{}` is a valid AS2/AP object.

**Comment by ShadowJonathan** [comment]
So, practically, what would this mean for an implementation like Mastodon?

Use `Annotation`, and send it via `Create` objects to remote inboxes?

Can/should it then fetch existing annotations via `/outbox`, when it first starts subscribing to a moderation list/auditor?

**Comment by ThisIsMissEm** [comment]
Let's focus first on how we'd want to support annotations, since there's many different approaches here.

**Comment by npdoty** [comment]
Is a separate Annotation type needed for any of institutional fact-checking, community-note context additions, or clarifications/corrections? Those seem like they all might be replies to a post that are intended to have a broad audience including those who have received or come across the original post.

Decentralized services won't have the luxury of declaring who the annotators/fact-checkers are, and it might be harder to construct a voted community consensus on a community note. But it might also be beneficial to recognize up front that different people will recognize different sources of fact checking and context addition. Maybe they can do that by having instances recommend (or people share lists) of accounts to subscribe to that only distribute fact-check follow-up-type response posts.

**Comment by jmking-iftas** [comment]
> Decentralized services won't have the luxury of declaring who the annotators/fact-checkers are, and it might be harder to construct a voted community consensus on a community note. But it might also be beneficial to recognize up front that different people will recognize different sources of fact checking and context addition. Maybe they can do that by having instances recommend (or people share lists) of accounts to subscribe to that only distribute fact-check follow-up-type response posts.

This is akin to the AT Proto model, anyone can stand up a "labeler" account and provide annotations, anyone can follow that labeler as an explicit means of "subscribing" to that account's annotations. IIRC the follow UI is a little different from a generic user follow, but pretty much the same idea.

**Comment by jmking-iftas** [comment]
> This is a rather interesting article on content warnings and labeling of content: https://shelraphen.com/on-content-warnings/

Just want to bookmark some relevant best practice input here:

 - https://sites.lsa.umich.edu/inclusive-teaching-sandbox/wp-content/uploads/sites/853/2021/02/An-Introduction-to-Content-Warnings-and-Trigger-Warnings-Draft.pdf
 -  https://languageplease.org/content-advisory-and-trigger-warnings/
 - https://provost.uconn.edu/trigger-warning/
 - https://pmc.ncbi.nlm.nih.gov/articles/PMC9067675/

**Comment by ThisIsMissEm** [comment]
So I think to move this forwards, we really need to split up the idea of Annotations and Labelling — these are very different use-cases — annotations are more for features like community notes or fact-checking; Labelling has it's own aspect which is self-applied labels vs third-party applied labels.

For content hosts, this is where I talk more of the notion of firewall based approaches to federation: Accept content, Reject it, silently Drop it, or Filter it. Filtering would not be an annotation or label, but rather just a change in the way the post is displayed to a user, at the Service Operators' discretion.

**Comment by gunchleoc** [comment]
Brain dump for features that might be useful for community notes:

# Annotation UI

* Community notes should be able to link to multiple sources
* Authors of community notes need to be able to edit/delete them in case they made an honest mistake
* What about dishonest mistake, e.g. annotating the sh*t out of somebody and delete everything after an hour or so? Should we log some stats about deletions here to refer to if somebody reports them?
* What about user safety when good-faith notes get federated to a malicious originating instance?

# Moderation interface

* Keep image hashes + notes from posts that have community notes, so that they can be flagged up on repeat. Same for links.
* Possibility to accept a note as official to be attached automatically - should it be anonymized then?
* Option to autoreport for moderation action (no automatic block, since somebody could be discussing that it's fake)?

# Frequent offenders

* List people who have been fact checked a lot for potential moderation action.
* Privacy/data protection: Record as much information as needed but not more than necessary about posts that have notes attached to them. Respect deleted posts.
* You can vet whether they're being dogpiled or whether they are spreading falsehoods. Could look a bit like Mastodon's trending hashtags UI. Such dogpiling should also be reportable.

# Trusted sources

* List of official/trusted fact checkers - in Germany, we have Correctiv and Volksverpetzer, for example.
* This could be accounts or other instances where you trust their moderation team.
* Do not list those people publicly without their consent (they're targets for harassment).
* Possibility to donate to people who do a lot of fact checks.

**Comment by ThisIsMissEm** [comment]
I'm going to lock this issue, as there's too many topics conflated into it, and that's partly my fault when I created the issue, I meant it in a very exploratory way. It looks like there is now consensus that we do what both labelling and annotations to exist.

Sometimes labels will be first-party (i.e., added by the publisher whether that's the actor themselves or an edit made by the instance admin), and there will be cases where they will be third-party, in which case an Annotation is more appropriate. It would be wrong to collapse these all down to Annotations though, as that could cause context collapse or confusion or difficult differentiating between what the publisher of the content has said vs what a third-party has said.
